### ISSUE TITLE
Confirm option for heroku_addon uses app id instead of app name.

### ISSUE THREAD

**Issue by ForgeRubenAguilar** [heroku/terraform-provider-heroku#334]
Running

```
Terraform v1.1.0
+ provider.heroku v5.0.1
```

### Affected Resource(s):
- heroku_addon

### Expected Behavior
Addon is created with need to confirm app name.

### Actual Behavior
Creating a database addon in an app inside a private space that requires confirmation as per [Heroku Private Space docs](https://devcenter.heroku.com/articles/private-spaces#add-ons-marked-as-available-in-a-private-space) gives the following error:
```
| Error: Post "https://api.heroku.com/apps/<REDACTED>/addons": This add-on is not automatically networked with this Private Space.
| in resource "heroku_addon" "database":
│   67: resource "heroku_addon" "database" {
│
```

### Steps to Reproduce
A `heroku_app` in a `heroku_space` with a database `heroku_addon` should do it.

### Important Factoids
Looking through the code, it appears that during the 4.5 => 5.0 upgrade swapping out of all the app names to app ids ended up swapping the confirm option for `heroku_addon` resource [here.](https://github.com/heroku/terraform-provider-heroku/blob/v5.0.1/heroku/resource_heroku_addon.go#L121) So it tries to confirm with app id instead of app name and subsequently fails.

**Comment by mars** [comment]
Thanks for opening this clear issue @ForgeRubenAguilar 😄

Fortunately that's an easy fix: PR #335 is now running acceptance tests, and we'll await a collaborator approval before merging and releasing.

**Comment by mars** [comment]
Fix [published](https://registry.terraform.io/providers/heroku/heroku/latest) as `v5.0.2`.
